src/annoylib.h

```diff
@@ -457,7 +513,14 @@ template<typename S, typename T, typename Distance>
           children.push_back(n);
       }
 
-      Distance::create_split(children, _f, &_random, m);
+      {
+        // Sample two random points from the set of nodes so that i != j
```

Comment:
i think that's a leaky abstraction – what if at some point later another distance function has a very different way to create a split plane?